testah-junit/src/main/java/org/testah/framework/cli/Cli.java

```diff
@@ -241,40 +223,48 @@ public void processRun()
         int totalTestCasesIgnored = 0;
         int totalTestPlans = 0;
 
-        if (null != results)
-        {
+        if (null != results) {
             totalTestPlans = results.size();
             TS.log().info(Cli.BAR_LONG);
             TS.log().info(Cli.BAR_WALL + "TestPlan Result(s):");
 
             TS.util().pause(1000L, "Waiting for TestsPlans to complete");
-            for (final ResultDto result : results)
-            {
+            for (final ResultDto result : results) {
 
-                if (null != result.getTestPlan())
-                {
+                if (null != result.getTestPlan()) {
                     result.getTestPlan().getRunInfo().recalc(result.getTestPlan());
                     totalTestCases += result.getTestPlan().getRunInfo().getTotal();
                     totalTestCasesFailed += result.getTestPlan().getRunInfo().getFail();
                     totalTestCasesPassed += result.getTestPlan().getRunInfo().getPass();
                     totalTestCasesIgnored += result.getTestPlan().getRunInfo().getIgnore();
-                } else
-                {
-                    TS.log().error("Testplan is null, for " + result.getJunitResult().getFailures());
+                } else {
+                    TS.log().error("Testplan is null, looking at junit result for " + result.getJunitResult().getFailures());
+
+                    if (null != result.getJunitResult()) {
+                        if (null != result.getJunitResult().getFailures()) {
+                            result.getJunitResult().getFailures().forEach(failure -> {
+                                if (StringUtils.startsWithIgnoreCase(failure.toString(), "initializationError")) {
```

Comment:
errors are not counted unless they are initialization errors?

Reply:
There can be, but will be accounted for in the junit result object used to fill in the metrics. The really bad failure that the framework has issues with since it never gets engaged is the class initialization failure.